packages/block-library/src/navigation-link/edit.js

```diff
@@ -196,13 +204,26 @@ function NavigationLinkEdit( {
 
 export default compose( [
 	withSelect( ( select, ownProps ) => {
-		const { getClientIdsOfDescendants, hasSelectedInnerBlock } = select( 'core/block-editor' );
+		const {
+			getBlockName,
+			getBlockAttributes,
+			getBlockParents,
+			getClientIdsOfDescendants,
+			hasSelectedInnerBlock,
+		} = select( 'core/block-editor' );
 		const { clientId } = ownProps;
+		const rootBlock = getBlockParents( clientId )[ 0 ];
+		const parentBlock = getBlockParents( clientId, true )[ 0 ];
+		const rootBlockAttributes = getBlockAttributes( rootBlock );
+		const hasDescendants = !! getClientIdsOfDescendants( [ clientId ] ).length;
+		const isLevelZero = getBlockName( parentBlock ) === 'core/navigation';
+		const showSubmenuIcon = rootBlockAttributes.showSubmenuIcon && isLevelZero && hasDescendants;
```

Comment:
It's nice to see that exists a way to access attributes of the parent block; however, I'm not sure if it's recommended. It seems that this feature has been deprecated. Pinging to @youknowriad in order to confirm it.

Reply:
going forward and in case we can't use `rootBlock`, I wonder if we could add the chevron icon using CSS. Maybe defining the icon in a pseudo-class?

Reply:
can we consider to replace the SVG with a char? Using `›`, for instance? cc @mtias 
It will make the implementation super easy and really simpler.
Even the approach of using borders is a nice option, [just using CSS](https://github.com/WordPress/gutenberg/pull/19681/commits/91a7300f506bad8a3f42a78e7e14db86441b407c):

![image](https://user-images.githubusercontent.com/77539/72610065-3977e580-3905-11ea-8df9-8d0fc15821ea.png)

Reply:
Researching a little bit, I think we can use a dashicon here, @WunderBart https://developer.wordpress.org/resource/dashicons/#arrow-down-alt2

Reply:
Thanks for pointing to Dashicons, @retrofox! I chose that chevron `SVG` as it's used in other menus (like block settings on the right-hand side) so it felt natural to me.

Reply:
> I chose that chevron SVG as it's used in other menus (like block settings on the right-hand side) so it felt natural to me.

I think Dashicons is better than adding an SVG.

> Are Dashicons going to be used in the submenu's current redesign work?

We can stop working on adding the arrow in https://github.com/WordPress/gutenberg/pull/19681, in order to avoid effort overlapping. cc @frontdevde

Reply:
The svg is fine, icons in general are in flux right now. Also, dashicons were meant to be used in the dashboard. Since these icons are displayed in the front of the site, a custom svg / image seems better.

Reply:
Good point. 🙇 Thanks!

Reply:
> We can stop working on adding the arrow in #19681, in order to avoid effort overlapping

Good idea. Probably worth taking note that in this PR, I want to address only the top-level navigation items (as the setting description says: `Show submenu icon for top-level items`). IMO it would be good to have this one shipped before the [new submenu design work](https://github.com/WordPress/gutenberg/pull/19681) one to address any visual inconsistencies there.